cleanup_utils/aws_cleanup.py

```diff
@@ -0,0 +1,227 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015-2018 OpenCraft <contact@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+S3 Cleanup Script
+
+Cleans up all AWS IAM Users, Policies, Access Keys and Buckets
+"""
+
+import logging
+import boto3
+import botocore
+
+
+# Constants ###################################################################
+
+DEFAULT_POLICY_NAME = "allow_access_s3_bucket"
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger('integration_cleanup')
+
+
+# Classes #####################################################################
+
+class AwsCleanupInstance:
+    """
+    Handles the cleanup of IAM users, Policies and Buckets related to CircleCI
+    """
+    def __init__(self, age_limit, aws_access_key_id, aws_secret_access_key, dry_run=False):
+        """
+        Set's up AWS connections and clients
+        """
+        self.age_limit = age_limit
+        self.dry_run = dry_run
+        self.cleaned_up_hashes = []
+
+        self.session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+        )
+        self.iam_client = self.session.client('iam')
+        self.s3_client = self.session.client('s3')
+        self.s3_resource = self.session.resource('s3')
+
+    def delete_bucket(self, bucket_name):
+        """
+        Deletes a S3 bucket and all of it's files
+        """
+        if self.dry_run:
+            return
+        try:
+            logger.info("Deleting bucket %s.", bucket_name)
+            bucket = self.s3_resource.Bucket(bucket_name)
+            bucket.object_versions.delete()
+            bucket.objects.all().delete()
+            bucket.delete()
+        except self.s3_client.exceptions.NoSuchBucket:
+            # Ignore if the bucket doesn't exist
+            logger.warning("The bucket %s did not exist, skipping.", bucket_name)
+        except botocore.exceptions.ClientError as e:
+            # Ignore if the bucket doesn't exist
```

Comment:
@giovannicimolin, this comment seems out of place since the 'bucket doesn't exist' exception is handled in the previous `except` clause.